metrics/processors/kubeprocessors/pod_based_enricher.go

```diff
@@ -196,7 +201,19 @@ func intValue(value int64) core.MetricValue {
 	}
 }
 
-func NewPodBasedEnricher(podLister *cache.StoreToPodLister) (*PodBasedEnricher, error) {
+func NewPodBasedEnricher(url *url.URL) (*PodBasedEnricher, error) {
```

Comment:
Ok, It seems the biggest problem in my pr.  I'll try to create a public cache repository for getting cache store or indexer.